test/saithrift/tests/copp_tests.py

```diff
@@ -0,0 +1,549 @@
+class LACPTest(NoPolicyTest):
+    def __init__(self):
+        NoPolicyTest.__init__(self)
+
+    def runTest(self):
+        self.run_suite()
+
+    def contruct_packet(self, port_number):
+        packet = simple_eth_packet(
+               pktlen=14,
+               eth_dst='01:80:c2:00:00:02',
+               eth_type=0x8809
+               ) / (chr(0x01)+(chr(0x01)))
+
+        return packet
+
+# SNMP packets are trapped as IP2ME packets.
+# IP2ME configuration in SONIC contains policer 
+class SNMPTest(PolicyTest): #FIXME: trapped as ip2me. mellanox should add support for SNMP trap
+    def __init__(self):
+        PolicyTest.__init__(self)
+
+    def runTest(self):
+        self.run_suite()
+
+    def contruct_packet(self, port_number):
+        src_mac = self.mac_map[port_number]
+        dst_mac = self.test_params['dst_mac']
+        dst_ip = self.peerip[port_number]
+        packet = simple_udp_packet(
+                          eth_dst=dst_mac,
+                          ip_dst=dst_ip,
+                          eth_src=src_mac,
+                          udp_dport=161
+                          )
+        return packet
+
+# SONIC configuration has no policer limiting for SSH
+class SSHTest(PolicyTest): # FIXME: ssh is policed now
+    def __init__(self):
+        PolicyTest.__init__(self)
+
+    def runTest(self):
+        self.run_suite()
+
+    def contruct_packet(self, port_number):
+        dst_mac = self.test_params['dst_mac']
+        src_ip = self.myip[port_number]
+        dst_ip = self.peerip[port_number]
+
+        packet = simple_tcp_packet(
+                eth_dst=dst_mac,
+                ip_dst=dst_ip,
+                ip_src=src_ip,
+                tcp_sport=22,
+                tcp_dport=22)
+
+        return packet
+
+# IP2ME configuration in SONIC contains policer 
+class IP2METest(PolicyTest):
+    def __init__(self):
+        PolicyTest.__init__(self)
+
+    def runTest(self):
+        self.run_suite()
+
+    def one_port_test(self, port_number, packet_count):
+        for i in xrange(self.MAX_PORTS):
+            packet = self.contruct_packet(i)
+            total_rcv_pkt_cnt, time_delta, time_delta_ms, tx_pps, rx_pps = self.copp_test(packet, packet_count, (0, port_number), (1, port_number))
+            self.printStats(self.PKT_TX_COUNT, total_rcv_pkt_cnt, time_delta, tx_pps, rx_pps)
+            self.check_constraints(total_rcv_pkt_cnt)
+
+        return
+
+    def contruct_packet(self, port_number):
+        src_mac = self.mac_map[port_number]
+        dst_mac = self.test_params['dst_mac']
+        dst_ip = self.peerip[port_number]
+
+        packet = simple_tcp_packet(
+                      eth_src=src_mac,
+                      eth_dst=dst_mac,
+                      ip_dst=dst_ip
+                      )
+
+        return packet
+
+
+class DefaultTest(PolicyTest):
+    def __init__(self):
+        PolicyTest.__init__(self)
+
+    def runTest(self):
+        self.run_suite()
```

Comment:
In this case, you need to setup the ttl error trap, setup route, and send packt.